include/swift/ABI/TaskOptions.h

```diff
@@ -0,0 +1,97 @@
+//===--- TaskOptions.h - ABI structures for task options --------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Swift ABI describing task options.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_ABI_TASK_OPTIONS_H
+#define SWIFT_ABI_TASK_OPTIONS_H
+
+#include "swift/ABI/TaskLocal.h"
+#include "swift/ABI/Executor.h"
+#include "swift/ABI/HeapObject.h"
+#include "swift/ABI/Metadata.h"
+#include "swift/ABI/MetadataValues.h"
+#include "swift/Runtime/Config.h"
+#include "swift/Basic/STLExtras.h"
+
+namespace swift {
+
+// ==== ------------------------------------------------------------------------
+// ==== Task Options, for creating and waiting on tasks
+
+/// The abstract base class for all options that may be used
+/// to configure a newly spawned task.
+class TaskOptionRecord {
+public:
+  TaskOptionRecordFlags Flags;
+  TaskOptionRecord *Parent;
+
+  TaskOptionRecord(TaskOptionRecordKind kind,
+                   TaskOptionRecord *parent = nullptr)
+  : Flags(kind) {
+    Parent = parent;
+  }
+
+  TaskOptionRecord(const TaskOptionRecord &) = delete;
+  TaskOptionRecord &operator=(const TaskOptionRecord &) = delete;
+
+  TaskOptionRecordKind getKind() const {
+    return Flags.getKind();
+  }
+
+  TaskOptionRecord *getParent() const {
+    return Parent;
+  }
+
+};
+
+/******************************************************************************/
+/****************************** TASK OPTIONS **********************************/
+/******************************************************************************/
+
+class TaskGroupTaskOptionRecord : public TaskOptionRecord {
+  TaskGroup *Group;
+
+  public:
+    TaskGroupTaskOptionRecord(TaskGroup *group)
+    : TaskOptionRecord(TaskOptionRecordKind::TaskGroup),
+      Group(group) {}
+
+  TaskGroup *getGroup() const {
+    return Group;
+  }
+};
+
+
+/// Task option to specify on what executor the task should be executed.
+///
+/// Not passing this option implies that that a "best guess" or good default
+/// executor should be used instead, most often this may mean the global
+/// concurrent executor, or the enclosing actor's executor.
+class ExecutorTaskOptionRecord : public TaskOptionRecord {
+  ExecutorRef *Executor;
```

Comment:
I'm pretty sure you want `ExecutorRef` rather than `ExecutorRef *` here.

Reply:
Oh, okey, will do

Reply:
Nah, I got it.